src/sync/incomingSyncUtils.js

```diff
@@ -12,117 +12,244 @@ import {
 
 import { SETTINGS_KEYS } from '../settings';
 const { THIS_STORE_ID } = SETTINGS_KEYS;
-
 import { CHANGE_TYPES, generateUUID } from '../database';
+import { setPriority } from 'os';
+import { PassThrough } from 'stream';
 
 /**
- * Take the data from a sync record, and integrate it into the local database as
- * the given recordType. If create or update, will update an existing record if
+ * Take the data from a sync object, and integrate it into the local database as
+ * the given objectType. If create or update, will update an existing object if
  * an id matches, or create a new one if not. If delete, will just clean up/delete.
  * @param  {Realm}  database   The local database
  * @param  {object} settings   Access to app settings
- * @param  {string} syncType   The type of change that created this sync record
- * @param  {object} syncRecord Data representing the sync record
+ * @param  {string} syncType   The type of change that created this sync object
+ * @param  {object} syncobject Data representing the sync object
  * @return {none}
  */
-export function integrateRecord(database, settings, syncRecord) {
-  // If the sync record is missing either data, record type, sync type, or record ID, ignore
-  if (!syncRecord.RecordType || !syncRecord.SyncType) return;
-  const syncType = syncRecord.SyncType;
-  const recordType = syncRecord.RecordType;
+export function integrateobject(database, settings, syncobject) {
+  // If the sync object is missing either data, object type, sync type, or object ID, ignore
+  if (!syncobject.objectType || !syncobject.SyncType) return;
+  const syncType = syncobject.SyncType;
+  const objectType = syncobject.objectType;
   const changeType = SYNC_TYPES.translate(syncType, EXTERNAL_TO_INTERNAL);
-  const internalRecordType = RECORD_TYPES.translate(recordType, EXTERNAL_TO_INTERNAL);
+  const internalobjectType = object_TYPES.translate(objectType, EXTERNAL_TO_INTERNAL);
+  if (changeType === 'merge') {
```

Comment:
this can be in switch statement below

Reply:
Logic for detection of merge sync record moved into relevant switch statement.